scripts/list_codeblocks.py

```
@@ -0,0 +1,213 @@
+
+# Pattern to match code blocks: ```language (optional) followed by code and closing ```
+CODEBLOCK_PATTERN = re.compile(
+    r'```(\w+)?\n(.*?)```',
+    re.DOTALL | re.MULTILINE
+)
+
+# Pattern to match inline code blocks (single backticks)
+INLINE_CODE_PATTERN = re.compile(r'`([^`]+)`')
+
+
+def extract_codeblocks(content: str, filepath: Path) -> List[Dict]:
```

Ruff (0.14.10)

Unused function argument: `filepath`

(ARG001)